app/controllers/api/v1/allocation_events_controller.rb

```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class AllocationEventsController < ApiController
```

Comment:
minor (non-blocking) point: it would be best to validate the parameters (especially that `timestamp` is in an ISO-8601 format), to prevent taking liberties with date formats. See e.g.  `app/services/journeys/params_validator.rb`